perf_events.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2019 Dropbox, Inc.
+// Full license can be found in the LICENSE file.
+
+package goebpf
+
+/*
+#ifdef __linux__
+#include <linux/perf_event.h>
+#define PERF_EVENT_HEADER_SIZE		(sizeof(struct perf_event_header))
+#else
+// mocks for Mac
+#define PERF_EVENT_HEADER_SIZE		8
+#define PERF_RECORD_SAMPLE			9
+#define PERF_RECORD_LOST			2
+#endif
+
+*/
+import "C"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"sync"
+)
+
+// PerfEvent is a way to interact with Linux's PerfEvents for eBPF case.
+type PerfEvents struct {
+	// Statistics
+	EventsReceived   int
+	EventsLost       int
+	EventsUnknowType int
+
+	// PollTimeoutMs is timeout for blocking call of poll()
+	// Defaults to 100ms
+	PollTimeoutMs int
+
+	perfMap        Map
+	updatesChannel chan []byte
+	stopChannel    chan bool
```

Comment:
nit: as this chan is used only for signalling you might want to use `stopChannel chan struct{}` (0-sized object)

Reply:
I don't have any preference here, so changed to `chan struct{}` :)